xDSCResourceDesigner.psm1

```diff
@@ -2160,35 +2240,57 @@ function Test-SchemaProperty
             -ErrorId $errorId -ErrorAction Continue
         $ErrorIdsRef.Value +=  $errorId 
     }
-
+    else
+    {
+        $type = $TypeMap[$simplifiedType]
+    }
 
     if ($CimProperty.Qualifiers["ValueMap"] -or $CimProperty.Qualifiers["Values"])
     {
         $ValueMap = $CimProperty.Qualifiers["ValueMap"].Value
         $Values = $CimProperty.Qualifiers["Values"].Value
 
-        $error = $false
+        $foundError = $false
 
         # Make sure if either of ValueMap or Values are present, both are.
         if ((-not ($ValueMap -and $Values)) `
             -or ($ValueMap.Count -ne $Values.Count))
         {
-            $error = $true
+            $foundError = $true
         }
-        else
+        elseif ($simplifiedType -like 'String*')
         {
+            # We only do this test for String properties.  Enums that map string names to numeric values (and so don't have
+            # identical values / valuemap arrays) are acceptable.
+
+            # TODO:  Should we also allow for valuemaps that map strings to other strings?  This is legal in WMI, though
+            # uncommon and perhaps not a use case that anyone cares about for DSC.
+
             for ($i = 0; $i -lt $ValueMap.Count; $i++)
             {
                 # Make sure the values contained in ValueMap and Values are identical
                 if ($ValueMap[$i] -ne $Values[$i])
                 {
-                    $error = $true
+                    $foundError = $true
                     break
                 }
             }
         }
-        
-        if ($error)
+        elseif ($null -ne $type)
+        {
+            foreach ($mappedValue in $ValueMap)
+            {
+                if ($null -eq ($mappedValue -as $type))
```

Comment:
Not sure what you mean.  If the conversion fails, the -as operator returns $null, which is what we're testing for in the conditional.  (I wouldn't be surprised if -as is just calling that ConvertTo() function under the hood, but I haven't checked.)